src/super_gradients/training/datasets/segmentation_datasets/segmentation_dataset.py

```diff
@@ -6,12 +6,18 @@
 from typing import Callable
 import torchvision.transforms as transform
 from PIL import Image
+
+from super_gradients.common.decorators.factory_decorator import resolve_param
+from super_gradients.common.factories.transforms_factory import TransformsFactory
 from super_gradients.training.datasets.sg_dataset import DirectoryDataSet, ListDataset
 from super_gradients.training.utils.segmentation_utils import RandomFlip, Rescale, RandomRotate, PadShortToCropSize, \
     CropImageAndMask, RandomGaussianBlur, RandomRescale
 
 
 class SegmentationDataSet(DirectoryDataSet, ListDataset):
+
+    @resolve_param('image_mask_transforms', factory=TransformsFactory())
```

Comment:
@shaydeci  do we really need both `image_mask_transforms` and `image_mask_transforms`, the actual one that is used is determined by `augment: bool`, the other is redundant.
IMO there should be only `image_mask_transforms`, `augment` is redundant and only complicating this dataset class.

Reply:
I personally don't mind it. I think it can come in handy if you want to cancel out augmentations at some point (for example fro debugging). Both ways are okay on my end.